js/channels/012.js

```diff
@@ -0,0 +1,173 @@
+            this.timePlus60.text(moment().startOf('hour').add(1, 'hours').format('h:mm A'));
+            this.date.text(moment().format('dddd MMMM D YYYY'));
+        }, 1000);
+
+        this.listingGrid.empty();
+        
+        guideData.find('channel').each((i, channel) => {
+            const ch = $(channel);
+            // channel notices
+            const notices = $(ch.find('notice'));
+            $(notices).each((i, notice) => {
+                const noticeRow = $('<tr>');
+                noticeRow.append($('<td>').addClass('notice').attr('colspan', 4).html($(notice).html()));
+                this.listingGrid.append(noticeRow);
+            });
+            
+            if (ch.attr('noticeonly')) return;
+            const channelRow = $('<tr>');
+            
+            const channelBox = $('<td>').addClass('channel');
+            channelBox.append($('<div>').addClass('number').text(ch.attr('number')));
+            channelBox.append($('<div>').addClass('station').text(ch.attr('name')));
+            channelRow.append(channelBox);
+            
+            const listings = $(ch.find('listing'));
+            
+            let listing1 = $(this.getFirstListing(listings, this.curTsStart));
+            let listing2 = $(this.getFirstListing(listings, this.curTsStart+1));
+            let listing3 = $(this.getFirstListing(listings, this.curTsStart+2));
+            
+            if (!listing1) {
+                listing1 = $('<listing>').html('Off-Air');
+            }
+            let showTwo = listing2 ? true : false;
+            if (listing1) showTwo = listing2 ? listing1.attr('timeslot') !== listing2.attr('timeslot') : false;
+            let showThree = listing3 ? true : false;
+            if (listing2) showThree = listing3 ? listing2.attr('timeslot') !== listing3.attr('timeslot') : false;
+            
+            if (showTwo) {
+                channelRow.append(this.generateListing(listing1));
+                if (!showThree) {
+                    channelRow.append(this.generateListing(listing2, 2));
+                } else {
+                    channelRow.append(this.generateListing(listing2));
+                    channelRow.append(this.generateListing(listing3));
+                }
+            } else {
+                if (!showThree) {
+                    channelRow.append(this.generateListing(listing1, 3));
+                } else {
+                    channelRow.append(this.generateListing(listing1, 2));
+                    channelRow.append(this.generateListing(listing3));
+                }
+            }
+            
+            this.listingGrid.append(channelRow);
+        });
+
+        // ads
+        guideData.find('ad').each((i, ad) => {
+            this.adList.push($(ad).html());
+        });
+
+        if (this.adList.length > 0) {
+            this.textLeft.html(this.adList[0]);
+        }
+
+        this.adInterval = setInterval(() => {
+            this.nextAd();
+        }, 30000);
```

Comment:
I'll set as an attribute in the guide XML, specifying a default value of 30 seconds otherwise.